add dcgm-exporter

yogiseven opened the pull request:
add dcgm-exporter in chart.

dongjiang1989 committed:
dongjiang, add loggie logger agent

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, add cset

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, add loggie agent

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, add promtail agent dashboard

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, fix grafana metrics

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, fix rules bug

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, add cilium dashboard

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, add feature readiness and liveness Probe

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, add alertmanager liveness/ readiness

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, fix bug

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, add csaert

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, add job template

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, fix bug

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, update image

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang update image version

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, update v

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
Update README.md

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, fix bug

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, add ebpf arm64 & amd64

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, update

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, add rule

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, update

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, add vpa dashboard, nf_conntrack panel and add Dropdown feature

Signed-off-by: dongjiang1989 <dongjiang1989@126.com>
Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, add vpa dashboard Dropdown feature

Signed-off-by: dongjiang1989 <dongjiang1989@126.com>
Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, remove empty dashboard by code review

Signed-off-by: dongjiang1989 <dongjiang1989@126.com>
Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
Update README.md

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, add dualStack

Signed-off-by: dongjiang1989 <dongjiang1989@126.com>
Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 committed:
dongjiang, fix bug

Signed-off-by: yogiseven <seventhyou@126.com>

yogiseven committed:
add dcgm-exporter

Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 commented:
@yogiseven  Fix DCO signoff

yogiseven committed:
Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 commented on metrics/Chart.yaml:
```diff
@@ -28,6 +28,10 @@ dependencies:
   name: grafana
   repository: https://grafana.github.io/helm-charts
   version: 6.43.*
+- condition: dcgmeEporter.enabled
```
`dcgmeEporter.enabled` not been defined in `value.yaml`

dongjiang1989 commented on metrics/Chart.yaml:
```diff
@@ -28,6 +28,10 @@ dependencies:
+- condition: dcgmeEporter.enabled
+  name: dcgm-exporter
+  repository: https://nvidia.github.io/dcgm-exporter/helm-charts
+  version: 3.1.5
```
regexp version for Kubernetes version v1.18 to v1.25+

dongjiang1989 commented on metrics/charts/dcgm-exporter/templates/service-monitor.yaml:
```diff
@@ -0,0 +1,42 @@
+# limitations under the License.
+
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
```
add grafana dashboard yaml

dongjiang1989 commented on metrics/charts/dcgm-exporter/templates/service.yaml:
```diff
@@ -0,0 +1,37 @@
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  type: {{ .Values.service.type }}
```
Add ipv4/ipv6 DualStack

dongjiang1989 commented on metrics/charts/dcgm-exporter/values.yaml:
```diff
@@ -0,0 +1,129 @@
+# limitations under the License.
+
+image:
+  repository: nvcr.io/nvidia/k8s/dcgm-exporter
```
change to  dockerhub.io

yogiseven committed:
Signed-off-by: yogiseven <seventhyou@126.com>

1, add dcgmeEporter.enabled  defined in value.yaml
2, add ipv4/ipv6 DualStack
3, change repository

Signed-off-by: yogiseven <seventhyou@126.com>

yogiseven committed:
update logging/loggie/Chart.yaml version

Signed-off-by: yogiseven <seventhyou@126.com>

yogiseven committed:
update  logging/loggie/values.yaml

Signed-off-by: yogiseven <seventhyou@126.com>

yogiseven committed:
update metrics/charts/prometheus-pingmesh-exporter/templates/service.yaml

Signed-off-by: yogiseven <seventhyou@126.com>

yogiseven committed:
update logging/loggie/values.yaml
Signed-off-by: yogiseven <seventhyou@126.com>

yogiseven committed:
sync metrics/values.yaml
Signed-off-by: yogiseven <seventhyou@126.com>

dongjiang1989 commented:
@yogiseven  need rebase code

yogiseven commented:
> @yogiseven need rebase code
rebase master done

yogiseven commented:
Do I need to submit a new PR, from yogiseven:master to kubeservice-stack:master？

dongjiang1989 commented:
> Do I need to submit a new PR, from yogiseven:master to kubeservice-stack:master？

You can close this PR. Then create new PR with real rebase version
